init.lua

```diff
@@ -147,18 +148,17 @@ function ranks.update_privs(player, trigger)
 end
 
 -- [function] Update player nametag
-function ranks.update_nametag(player)
+function ranks.update_nametag(name)
 	if minetest.settings:get("ranks.prefix_nametag") == "false" then
 		return
 	end
 
-	if type(player) == "string" then
-		player = minetest.get_player_by_name(player)
+	if type(name) == "string" then
+		player = minetest.get_player_by_name(name)
```

Comment:
Again this should be effectively flipped:
```lua
if type(name) ~= "string" then
    name = name:get_player_name()
```

Reply:
See line *169*. Does this still need to be implemented in some form?

Reply:
If I understand your question correctly, no, it doesn't need to be implemented as it is. But it does need to be flipped as I mentioned in my first comment. I'll address the other part of this in the comment on line 169.

Reply:
Done!